Backend file

jacobweinstock opened the pull request:
## Description

<!--- Please describe what this PR is going to change -->
This adds a file watcher backend. Useful for local development and testing.

## Why is this needed
Helps with local development and testing.

<!--- Link to issue you have raised -->

Fixes: #

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->


## How are existing users impacted? What migration steps/scripts do we need?

<!--- Fixes a bug, unblocks installation, removes a component of the stack etc -->
<!--- Requires a DB migration script, etc. -->


## Checklist:

I have:

- [X] updated the documentation and/or roadmap (if required)
- [X] added unit or e2e tests
- [X] provided instructions on how to upgrade

jacobweinstock committed:
Add file backend and example main:

This provides a local file watcher backend.
This backend's purpose is main for local development.

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>

jacobweinstock committed:
Add helper funcs for data.* to otel attributes:

These make it easier for all backends to consume.

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>

jacobweinstock committed:
Enable otel for the example/main.go

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>

jacobweinstock committed:
Add test for bad data

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>

jacobweinstock committed:
Update to use new EncodeTo methods

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>

jacobweinstock committed:
Add missing import

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>

jacobweinstock committed:
Add doc for file watcher backend.

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>

displague commented on example/main.go:
```diff
@@ -0,0 +1,55 @@
+package main
```
Let's move this to `examples/something/`

jacobweinstock replied:
sure thing. updated.

jacobweinstock committed:
Move file backend example to its own dir:

This will improve clarity of which backend example this is
With more than one backend examples available.

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>

jacobweinstock commented:
Hey @nshalman, you are the only approver in this repo (I need to change this), would you mind checking this out?
@displague, pinging you for another review, please.

nshalman commented on backend/file/file.go:
```diff
@@ -0,0 +1,253 @@
+
+				return nil, nil, err
+			}
+			span.SetAttributes(d.EncodeToAttributes()...)
```
This seems kind of wrong to me inside the loop. What are we trying to do here?

nshalman reviewed (approved):
This seems fine to me.
I didn't read the tests very closely, though.

jacobweinstock commented:
> This seems fine to me. I didn't read the tests very closely, though.

Thank you!
